File: 0075-sort-colors/0075-sort-colors.py
class Solution(object):
    def sortColors(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        i=0
        j=0
        k=len(nums)-1
        while(j<=k):
            print(nums,i,j,k)
            if nums[i]==0 and nums[j]==0 and i==j:
                i+=1
                j+=1
            elif(nums[j]==0):
                nums[j]=nums[i]
                nums[i]=0
                i+=1
            elif nums[j]==2:
                nums[j]=nums[k]
                nums[k]=2
                k-=1
            else:
                j+=1
            

        

File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int i=0;
        int j=0;
        int k=nums.length-1;
        while(i<=k){
            if(nums[i]==0){
                if(i==j){
                    i++;
                    j++;
                }
                else{
                    nums[j]=0;
                    j++;
                    nums[i]=1;
                    i++;
                }
            }
            else if(nums[i]==2){
                nums[i]=nums[k];
                nums[k]=2;
                k--;
            }else i++;
           
        }
    }
}